### ISSUE TITLE
Allow a package to get the version of itself

### ISSUE THREAD

**Issue by fjarri** [JuliaLang/Pkg.jl#385]
In julia 0.6 it was possible to call `Pkg.installed()["PkgName"]` inside a package `PkgName` and get the version number of the currently running package (e.g. to document in logs). In `Pkg3`, it seems, the versions returned by `installed()` are limited to the packages in `Packages.toml` or `Manifest.toml`, and do not include the calling package. Would it be possible to add an option of getting your own version (either include it in the existing output, or request it with a separate `PackageMode` value)?

**Comment by KristofferC** [comment]
Hm, but you know the version of the package itself? When packages have `Project.toml` files they will have a `version` entry that you can look at to know what version they are.

But yes, we need a better way than currently for querying different things about the current project. However, only the things in `Manifest.toml` are what is considered installed for that specific project so querying things outside that doesn't make so much sense.

**Comment by fjarri** [comment]
> When packages have Project.toml files they will have a version entry that you can look at to know what version they are.

When you say "you", what do you mean exactly? Certainly, I, as a person, know the version already. I would like to have a way to query it programmatically. Moreover, the function that could do that (`Pkg.installed()`) is already present, except it only provides information about the dependencies of my project, not about the project itself.

**Comment by DilumAluthge** [comment]
It shouldn't be too difficult to write a function in your package to read the `Project.toml` file, parse it, and extract the `version` entry. For example, you could create a file `MyPackage.jl/src/get_version.jl` that would have the following contents:
```julia
import Pkg

function get_my_version()
    version_number = try
    projecttoml_filename = joinpath(dirname(dirname(@__FILE__)),"Project.toml")
    projecttoml_parsed = Pkg.TOML.parse(read(projecttoml_filename, String))
    VersionNumber(projecttoml_parsed["version"])
    catch e
        @warn(string("Ignoring error: ", e))
        VersionNumber(0)
    end
    return version_number
end
```

**Comment by fjarri** [comment]
That's true, if I was blocked by this issue, I would probably do that. But in the end it is a hack. `Pkg` manages project files, and it can extract the version from `Project.toml`, it just doesn't have a way to obtain it when you are *inside* the project. This seems counterintuitive to me. Would it somehow go against the package philosophy to have that option?

**Comment by StefanKarpinski** [comment]
No, it would be fine to make the package version accessible within the package, it's just another feature to design and implement. Before hacking in just the version number, however, consider that that's only one bit of metadata about a package that we might want to expose. There should be a more general way of getting that metadata.

**Comment by fjarri** [comment]
I am content with waiting for a proper API for that. The thing that I don't quite understand is that this feature seems to be *already* designed and implemented, because outside the project you can call `Pkg.installed()["MyPackage"]` and get the version. Is this somehow different from doing it inside the project?

**Comment by StefanKarpinski** [comment]
Just a name is not sufficient to identify a package anymore, so that legacy API is inherently broken.

**Comment by 00vareladavid** [comment]
Resolved by #1086. The command would be `Pkg.project().version`
